test/puppetlabs/puppetdb/testutils/log_test.clj

```diff
@@ -0,0 +1,77 @@
+(ns puppetlabs.puppetdb.testutils.log-test
+  (:require [clojure.test :refer :all]
+            [clojure.tools.logging :as log]
+            [puppetlabs.kitchensink.core :as kitchensink]
+            [puppetlabs.puppetdb.testutils :refer [temp-file]]
+            [puppetlabs.puppetdb.testutils.log :as tgt])
+  (:import [org.slf4j Logger LoggerFactory]))
+
+(defn- event->map [event]
+  {:level (str (.getLevel event))
+   :message (.getMessage event)})
+
+(deftest with-log-level-and-logging-to-atom
+  (let [log (atom [])]
+    (tgt/with-log-level Logger/ROOT_LOGGER_NAME :error
+      (tgt/with-logging-to-atom Logger/ROOT_LOGGER_NAME log
+        (log/info "wlta-test"))
+      (is (not-any? #(= {:level "INFO" :message "wlta-test"} %)
+                    (map event->map @log)))))
+  (let [log (atom [])]
+    (tgt/with-log-level Logger/ROOT_LOGGER_NAME :info
+      (tgt/with-logging-to-atom Logger/ROOT_LOGGER_NAME log
+        (log/info "wlta-test"))
+      (is (some #(= {:level "INFO" :message "wlta-test"} %)
+                (map event->map @log))))))
+
+(deftest with-additional-log-appenders
+  ;; Definitely incomplete
+  (let [log (atom [])
+        uuid (kitchensink/uuid)]
+    (tgt/with-additional-log-appenders org.slf4j.Logger/ROOT_LOGGER_NAME
+      [(#'puppetlabs.puppetdb.testutils.log/log-event-listener
+        (fn [event] (swap! log conj event)))]
+      (log/error uuid))
+    (is (some #(= {:level "ERROR" :message uuid} %)
+              (map event->map @log)))))
+
+(deftest with-log-appenders
+  ;; Definitely incomplete
```

Comment:
I believe the main thing I meant there is that both of these tests only test the happy path.  They don't, for example, test that with-log-appenders clobbers all existing appenders, nor that with-additional-log-appenders doesn't.  I just realized that it might be fairly easy to add simple tests for those case.  I'll try.